jerry-core/ecma/base/ecma-globals.h

```diff
@@ -1121,6 +1133,22 @@ typedef struct
 
 #endif /* !CONFIG_ECMA_LCACHE_DISABLE */
 
+#ifndef CONFIG_DISABLE_TYPEDARRAY_BUILTIN
+
+/**
+ * Internal properties of TypedArray Object
+ */
+typedef struct
+{
+  /** The arraybuffer object behind the typedarray */
+  ecma_object_t *viewed_arraybuffer_p;
+  /** the byteoffset of the above arraybuffer */
+  ecma_length_t byte_offset;
+  /** the size of each element in the typedarray */
+  ecma_length_t bytes_per_element;
```

Comment:
Yes, we can do this.

About the 2 unused byte, do you mean the following part because of the 4-byte alignment?
```
    struct
    {
      uint16_t class_id; // 2 byte
      union
      {
        ecma_value_t value;
        uint32_t length;
      } u; //4 byte
    uint16_t unused_2_bytes;
    } class_prop;
```